docs/rancher/resource-quota.md

```diff
@@ -0,0 +1,75 @@
+    - `VirtLauncherOverhead`: 75Mi  (the `ps` RSS for the virt-launcher process)
+    - `VirtlogdOverhead`: 17Mi  (the `ps` RSS for virtlogd)
+    - `LibvirtdOverhead`: 33Mi. (the `ps` RSS for libvirtd)
+    - `QemuOverhead` : 30Mi (the `ps` RSS for qemu, minus the RAM of its (stressed) guest, minus the virtual page table)
+- 8Mi per CPU (vCPU) Overhead: Additionally, 8Mi of overhead per vCPU is added, along with a fixed 8Mi overhead for IOThread.
+- Extra Added Overhead: This encompasses various factors like video RAM overhead and architecture overhead. Refer to [Additional Overhead](https://github.com/kubevirt/kubevirt/blob/2bb88c3d35d33177ea16c0f1e9fffdef1fd350c6/pkg/virt-controller/services/template.go#L1853-L1890) for further details.
+
+This calculation demonstrates that the VM instance necessitates an additional memory overhead of approximately 276Mi.
+
+For more information, see [Memory Overhead](https://kubevirt.io/user-guide/virtual_machines/virtual_hardware/#memory-overhead).
+
+For more information on how the memory overhead is calculated in Kubevirt, refer to [kubevirt/pkg/virt-controller/services/template.go](https://github.com/kubevirt/kubevirt/blob/v0.54.0/pkg/virt-controller/services/template.go#L1804).
+
+## Adjust ResourceQuota Automatically during the Migration
+When the allocated resource quota controlled by the `ResourceQuota` object reaches its limit, migrating a VM becomes unfeasible. The migration process automatically creates a new pod mirroring the resource requirements of the source VM. If these pod creation prerequisites surpass the defined quota, the migration operation cannot proceed.
+
+_Available as of v1.2.0_
+
+In Harvester, the `ResourceQuota` values dynamically expand ahead of migration to accommodate the resource needs of the target virtual machine. After migration, the ResourceQuotas are reinstated to their prior configurations.
+
+Please be aware of the following considerations of the automatic resizing of `ResourceQuota`:
+- When raising the `ResourceQuota` value, if you create, start, or restore other VMs, Harvester will verify if the resources are sufficient based on the original `ResourceQuota`. If the conditions are not met, the system will alert that the migration process is not feasible.
+- `ResourceQuota` cannot be changed during VM migration.
+- After expanding `ResourceQuota`, potential resource contention may emerge between non-VM pods and VM pods, leading to migration failures. Consequently, deploying non-VM pods to the same namespace is not recommended.
+- Due to the concurrent limitation of the webhook validator, the VM controller will execute a secondary validation to confirm resource sufficiency. If the resources are insufficient, it will auto config VM's`RunStrategy` to **Halted**, and a new annotation `harvesterhci.io/insufficient-resource-quota: xxxxx` will be added to the VM object, informing you that the VM was shut down due to insufficient resources.
```

Comment:
```suggestion
- Due to the concurrent limitation of the webhook validator, the VM controller will execute a secondary validation to confirm resource sufficiency. If the resources are insufficient, it will auto config the VM's`RunStrategy` to **Halted**, and a new annotation `harvesterhci.io/insufficient-resource-quota: xxxxx` will be added to the VM object, informing you that the VM was shut down due to insufficient resources.
```